### PR TITLE
feat: Unsafe content support for DOM JVM.

### PR DESCRIPTION
**Pull request by gzoritchak**
Here is a proposal for https://github.com/Kotlin/kotlinx.html/issues/8 implementation.

### CODE REVIEW

File: jvm/src/main/kotlin/dom.kt [hunk 1]
```diff
@@ -79,7 +79,21 @@ class HTMLDOMBuilder(val document : Document) : TagConsumer<Element> {
     override fun finalize() = lastLeaved ?: throw IllegalStateException("No tags were emitted")
 
     override fun onTagContentUnsafe(block: Unsafe.() -> Unit) {
-        throw UnsupportedOperationException("unsafe content for DOM at JVM is not supported yet")
+        UnsafeImpl.block()
+    }
+
+    fun String.toElement() = DocumentBuilderFactory
```
Comment (cy6erGn0m) [inline jvm/src/main/kotlin/dom.kt]: can't we keep document builder in a lazy property?

File: jvm/src/main/kotlin/dom.kt [hunk 2]
```diff
@@ -79,7 +79,21 @@ class HTMLDOMBuilder(val document : Document) : TagConsumer<Element> {
...
+            .newInstance()
+            .newDocumentBuilder()
+            .parse(ByteArrayInputStream(this.toByteArray()))
```
Comment (cy6erGn0m) [inline jvm/src/main/kotlin/dom.kt]: what about character encoding here? shouldn't we configure document builder or use `StringReader` instead ?

**Comment by cy6erGn0m** [comment]
Great!

File: jvm/src/main/kotlin/dom.kt (hunk 1 above)
Comment (gzoritchak) [inline jvm/src/main/kotlin/dom.kt]: In case of multiple unsafe block in the same document? Yes.

File: jvm/src/main/kotlin/dom.kt (hunk 2 above)
Comment (gzoritchak) [inline jvm/src/main/kotlin/dom.kt]: Yes of course.

**Comment by cy6erGn0m** [comment]
What if `unsafe { +"prefix <a>tag</a> suffix" }` ?

**Comment by cy6erGn0m** [comment]
merge it for now: will fix issues later

**Pull request merged**